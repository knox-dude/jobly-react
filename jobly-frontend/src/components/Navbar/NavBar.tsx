import { useState } from 'react';
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
} from 'reactstrap';

function NavBar() {
  const args = {
    color: 'dark',
    light: false,
    dark: true,
    expand: "md",
    fixed: "top",
    toggle: "true",
  };
  const [isOpen, setIsOpen] = useState(false);

  const toggle = () => setIsOpen(!isOpen);

  return (
    <div>
      <Navbar {...args}>
        <NavbarBrand href="/">jobly</NavbarBrand>
        <NavbarToggler onClick={toggle} />
        <Collapse isOpen={isOpen} navbar>
          <Nav className="me-auto" navbar>
            <NavItem>
              <NavLink href="/companies/">Companies</NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="/companies/apple">
                Apple Company
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="/jobs">
                Jobs
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="/login">
                Login
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="/signup">
                Signup
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="/profile">
                Profile
              </NavLink>
            </NavItem>
            <UncontrolledDropdown nav inNavbar>
              <DropdownToggle nav caret>
                Options
              </DropdownToggle>
              <DropdownMenu end>
                <DropdownItem>Option 1</DropdownItem>
                <DropdownItem>Option 2</DropdownItem>
                <DropdownItem divider />
                <DropdownItem>Reset</DropdownItem>
              </DropdownMenu>
            </UncontrolledDropdown>
          </Nav>
        </Collapse>
      </Navbar>
    </div>
  );
}

export default NavBar;