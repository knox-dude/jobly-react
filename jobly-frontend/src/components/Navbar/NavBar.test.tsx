import { render, screen } from '@testing-library/react';
import NavBar from './NavBar';

// smoke test
test("should render without issue", () => {
  render(<NavBar />);
});

// snapshot test
test("matches snapshot", function() {
  const {asFragment} = render(<NavBar />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders NavBar with all links and options', () => {
  render(<NavBar />);

  expect(screen.getByText('jobly')).toBeInTheDocument();
  expect(screen.getByText('Companies')).toBeInTheDocument();
  expect(screen.getByText('Apple Company')).toBeInTheDocument();
  expect(screen.getByText('Jobs')).toBeInTheDocument();
  expect(screen.getByText('Login')).toBeInTheDocument();
  expect(screen.getByText('Signup')).toBeInTheDocument();
  expect(screen.getByText('Profile')).toBeInTheDocument();
  expect(screen.getByText('Options')).toBeInTheDocument();
  expect(screen.getByText('Option 1')).toBeInTheDocument();
  expect(screen.getByText('Option 2')).toBeInTheDocument();
  expect(screen.getByText('Reset')).toBeInTheDocument();
});
