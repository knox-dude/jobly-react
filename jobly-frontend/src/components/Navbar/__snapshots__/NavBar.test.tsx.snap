// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md navbar-dark bg-dark fixed-top"
      toggle="true"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          jobly
        </a>
        <button
          aria-label="Toggle navigation"
          class="navbar-toggler"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
        >
          <ul
            class="me-auto navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/companies/"
              >
                Companies
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/companies/apple"
              >
                Apple Company
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/jobs"
              >
                Jobs
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/login"
              >
                Login
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/signup"
              >
                Signup
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/profile"
              >
                Profile
              </a>
            </li>
            <li
              class="dropdown nav-item"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                class="dropdown-toggle nav-link"
                href="#"
              >
                Options
              </a>
              <div
                aria-hidden="true"
                class="dropdown-menu dropdown-menu-end"
                data-bs-popper="static"
                role="menu"
                tabindex="-1"
              >
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                >
                  Option 1
                </button>
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                >
                  Option 2
                </button>
                <div
                  class="dropdown-divider"
                  tabindex="-1"
                />
                <button
                  class="dropdown-item"
                  role="menuitem"
                  tabindex="0"
                >
                  Reset
                </button>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
